docs/_instructions/generate.md

```diff
@@ -0,0 +1,56 @@
+
+This `-generate` instruction specifies the code generating steps that must be executed. Source code can be generated by _system_ commands or the bnd _external plugins_.
+ 
+	-generate	::= 	clause ( ',' clause )*
+	clause		::= 	FILESET ';' 'output=' DIR (';' option )*
+	src		::=	FILESET
+	option		::= 	'system=' STRING
+			|   'generate=' STRING
+
+For each clause, the key of the clause is used to establish an Ant File Set, e.g. `foo/**.in`. This a glob expression with the exception that the double wildcard ('**') indicates to any depth of directories. The `output` attribute _must_ specify a directory. If the output must be compiled this directory must be on the bnd source path.
+
+If any file in the source is older than any file in the target (to any depth), or the target is empty, the clause is considered _stale_. If the clause is not stale, it is further ignored. If no further options are set on the clause, a warning is generated that some files are out of date. 
+
+If either a warning or error option is given, these will be executed on the project.
+
+If a command `STRING` is given it is executed as in the `${system}` macro. If the command `STRING` starts with a 
+minus sign (`-`) then a failure is not an error, it is reported as warning.
+
+The generate option will execute an _external plugin_ with the `objectClass` (service type) of `Generator<? extends Options>`. External plugins can come from an external repository or a local workspace project.
+
+The `generate` value is a _command line_. It can use the standard _unix_ like way of specifying a command. It supports flags (boolean parameters) and parameter that take a value. When this external plugin is executed, it is expected to create files fall within the _target_, if not, an error is reported. These changed or created files are refreshed in Eclipse.
+
+## Examples
+
+Include in the bnd build is a _javagen_ external plugin that is useful to generate Java code based on build information. It uses a template directory with Java files. When the external plugin runs, it will use all these files to write matching Java files in the output directory, in a matching package directory. The input Java files can be prefixed with a properties header:
+
+    ---
+    foo: 1
+    ---
+    package foobar;
+    class Foo {
+        int foo = ${foo};
+    }
+
+The optional header is removed and then the remainder of the file is ran through the bnd macro processor.
+
+If this example is used, it is necessary to add a new _source folder_. In Eclipse, this requires adding an entry in the `.classpath` file, in bnd it requires the modification of the `src` property.  
+
+    src=${^src},src-gen
+
+Assuming that the input files are in the `gen` directory, the following can be used to automatically generate the output files based on the input files.
+ 
+    -generate:   \
+        gen/**.java; \ 
+            output='src-gen/' ; \ 
+            generate='javagen -o src-gen gen/'
+
```

Comment:
Can you add an example, that an external Plugin? It is a but unclear how such an instruction would need too look like.

Reply:
https://github.com/bndtools/bnd/pull/3926/files#diff-3ea5fc0a86dd03fb518bf256100a9e27

Reply:
I'm not quite sure what the diff should show me.

Reply:
Sorry, was supposed to be a link to the external-plugins